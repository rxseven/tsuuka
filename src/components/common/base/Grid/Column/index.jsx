import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';

const propTypes = exact({
  children: PropTypes.node.isRequired,
  size: PropTypes.string
});

const defaultProps = {
  size: 'col'
};

function Column({ children, size }) {
  return <div className={size}>{children}</div>;
}

Column.propTypes = propTypes;
Column.defaultProps = defaultProps;

export default Column;
