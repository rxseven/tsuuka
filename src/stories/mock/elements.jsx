import React from 'react';

// Placeholder content
const content = <div>Placeholder content</div>;

export default { content };
