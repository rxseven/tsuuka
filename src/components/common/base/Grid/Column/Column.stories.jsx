import React from 'react';
import { storiesOf } from '@storybook/react';

import mock from 'stories/mock';
import notes from 'stories/shared/notes';
import Column from './index';

const { content } = mock.elements;
const title = 'Components/Common/Base/Grid/Column';
const component = <Column>{content}</Column>;

storiesOf(title, module).add('default', () => component, {
  notes: notes.minimal
});
