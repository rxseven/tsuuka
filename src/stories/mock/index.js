import elements from './elements';
import text from './text';

export default {
  elements,
  text
};
