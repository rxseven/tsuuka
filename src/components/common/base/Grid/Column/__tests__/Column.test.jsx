import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import mock from 'tests/mock';
import Column from '../index';

describe('<Column />', () => {
  // Arrange
  const props = { children: mock.elements.children };
  const component = <Column {...props} />;

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
