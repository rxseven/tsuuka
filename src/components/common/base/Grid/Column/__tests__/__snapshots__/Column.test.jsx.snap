// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Column /> Snapshot tests should render correctly 1`] = `
<div
  className="col"
>
  <div>
    Content
  </div>
</div>
`;
